src/stores/AccountsStore.js

```diff
@@ -118,36 +112,38 @@ export default class AccountsStore extends Container {
       const accountToSave = this.deleteSensitiveData(account);
 
       accountToSave.updatedAt = new Date().getTime();
-      await saveAccount(accountToSave);
+      await saveAccount(accountKey, accountToSave);
     } catch (e) {
       console.error(e);
     }
   }
 
 
-  async deleteAccount(account) {
+  async deleteAccount(accountKey) {
     const { accounts } = this.state;
 
-    accounts.delete(accountId(account));
+    accounts.delete(accountKey);
     this.setState({ accounts });
```

Comment:
Though currently how we deal the selectedKey is safe, I suggest to reset the `selectedKey` to `""` after we delete it to avoid potential problem.